File: 102_binaryTreeLevelOrder.cpp
vector<vector<int>> levelOrder(TreeNode* root)
{
	vector<vector<int>> res;
	if(root == NULL)
		return res;
	queue<TreeNode*> q;
	vector<int> path;
	q.push(root);
	int n = q.size();
	while(!q.empty())
	{
		n = q.size();
		while(n>0)
		{
			TreeNode* node = q.front();
			q.pop();
			path.push_back(node->val);
			if(node->left != NULL)
				q.push(node->left);
			if(node->right != NULL)
				q.push(node->right);
			n--;
		}
		res.push_back(path);
		path.clear();
	}
	return res;
}

File: 102_binaryTreeLevelOrder.java
public List<List<Integer>> levelOrder(TreeNode root)
{
	List<List<Integer>> res = new ArrayList<>();
	if(root == null)
		return res;
	Queue<TreeNode> q = new LinkedList<>();
	q.offer(root);
	while(!q.isEmpty())
	{
		int n = q.size();
		List<Integer> tmp = new ArrayList<>();
		for(int i=0; i<=n-1; i++)
		{
			TreeNode node = q.poll();
			tmp.add(node.val);
			if(node.left != null)
				q.offer(node.left);
			if(node.right != null)
				q.offer(node.right);
		}
		res.add(tmp);
	}
	return res;
}